File: UIViewControllerTransitions/Classes/DragDropTransition.m
//
//  DragDropTransition.m
//  UIViewControllerTransitions
//
//  Created by Steve Kim on 5/12/16.
//  Modified by Steve Kim on 4/14/17.
//      - Renew design and add new feature interactive transition
//      - Rename UIViewControllerDragDropTransition to DragDropTransition
//  Modified by Steve Kim on 8/14/17.
//      - Refactoring extract methods
//

#import "DragDropTransition.h"
#import "UIViewControllerTransitionsMacro.h"

@implementation DragDropTransition

#pragma mark - Overridden: UIViewControllerTransition

- (void)initProperties {
    [super initProperties];
    
    self.animationOptionsForDismission = self.animationOptionsForPresenting = 7;
    _imageViewContentMode = UIViewContentModeScaleAspectFill;
}

- (AnimatedTransitioning *)transitioningForDismissedController:(UIViewController *)dismissed {
    DragDropTransitioning *transitioning = [DragDropTransitioning new];
    transitioning.imageViewContentMode = _imageViewContentMode;
    transitioning.source = _dismissionSource;
    return transitioning;
}

- (AnimatedTransitioning *)transitioningForForPresentedController:(UIViewController *)presented presentingController:(UIViewController *)presenting sourceController:(UIViewController *)source {
    DragDropTransitioning *transitioning = [DragDropTransitioning new];
    transitioning.imageViewContentMode = _imageViewContentMode;
    transitioning.source =_presentingSource;
    return transitioning;
}

@end

File: AnimatedTransitionKit/Classes/Transitions/DragDrop/DragDropTransition.swift
//
//  DragDropTransition.swift
//  Pods
//
//  Created by Minwoo on 9/3/25.
//

import UIKit

public final class DragDropTransition: AnimatedTransition {
    var imageViewContentMode: UIView.ContentMode = .scaleAspectFill
    public var dismissionSource: DragDropTransitioningSource?
    public var presentingSource: DragDropTransitioningSource?
    
    // MARK: - Overridden: AnimatedTransition
    
    override public func initProperties() {
        super.initProperties()
        appearenceOptions.animationOptions = UIView.AnimationOptions(rawValue: 7)
        disappearenceOptions.animationOptions = UIView.AnimationOptions(rawValue: 7)
        appearenceOptions.isUsingSpring = true
        disappearenceOptions.isUsingSpring = true
        imageViewContentMode = .scaleAspectFill
    }
    
    override public func transitioning(forDismissedController dismissed: UIViewController?) -> AnimatedTransitioning? {
        let transitioning = DragDropTransitioning()
        transitioning.imageViewContentMode = imageViewContentMode
        transitioning.source = dismissionSource
        return transitioning
    }
    
    override public func transitioningFor(forPresentedController presented: UIViewController?, presenting: UIViewController?, sourceController source: UIViewController?) -> AnimatedTransitioning? {
        let transitioning = DragDropTransitioning()
        transitioning.imageViewContentMode = imageViewContentMode
        transitioning.source = presentingSource
        return transitioning
    }
}
